frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -0,0 +1,152 @@
+};
+
+// not going to type this yet, as this function will likely go away when we bring in fetch by provider
+// returns a list of all lookup tables to be filtered
+const getLookupTables = () =>
+    axios(lookupTableApi.getTableList()).then(({ data }) => data);
+
+/*
+
+  Helper function to find the table we want within the response for ALL tables
+  Hopefully this will go away with the API refactor
+
+*/
+const findTableByName = (
+    tables: LookupTable[],
+    tableName: LookupTables
+): TableAttributes => {
+    const filteredBody: LookupTable[] = tables.filter(
+        (tv: LookupTable) => tv.tableName === tableName && tv.isActive
+    );
+
+    if (!filteredBody.length) {
+        throw new Error(`Table '${tableName}' was not found!`);
+    }
+    const table: LookupTable = filteredBody.sort(
+        (a: LookupTable, b: LookupTable) =>
+            b["tableVersion"] - a["tableVersion"]
+    )[0];
+
+    const { tableVersion, createdAt, createdBy } = table;
+    if (tableVersion === undefined) {
+        throw new Error(`No version of table '${tableName}' was found!`);
+    }
+
+    return {
+        version: tableVersion,
+        createdAt,
+        createdBy,
+    };
+};
+
+/*
+
+  useQuery function
+
+  hook used to get value sets and value set rows (defined by passsed dataTableName)
+
+*/
+
+export const useValueSetsTable = <T extends ValueSet | ValueSetRow>(
+    dataTableName: LookupTables,
+    suppliedVersion?: number
+): {
+    valueSetArray: Array<T>;
+    error: any;
+} => {
+    let versionData: Partial<TableAttributes> | null;
+    let error;
+    let valueSetArray;
+
+    // get all lookup tables
+    const { error: tableError, data: tableData } = useQuery<LookupTable[]>(
+        ["lookupTables"],
+        () => getLookupTables(),
+        { enabled: !suppliedVersion } // only if version was not already passed in
+    );
+
+    // handle complexity around passed version vs. incomplete useQuery vs. success
+    if (suppliedVersion) {
+        versionData = {
+            version: suppliedVersion,
+        };
+    } else if (!tableData) {
+        versionData = null;
+    } else {
+        try {
+            versionData = findTableByName(tableData, dataTableName);
+        } catch (e) {
+            error = e;
+            versionData = null;
+        }
+    }
+
+    // unclear how to resolve the issue of defining a function to account for the absence of
+    // version when the `enabled` param should take care of not running if that's not present...
+    // get value for lookup table version
+
+    // not entirely accurate typing. What is sent back by the api is actually ApiValueSet[] rather than ValueSet[]
+    // does not seem entirely worth it to add the complexity needed to account for that on the frontend, better
+    // to make the API conform better to the frontend's expectations. TODO: look at this when refactoring the API
+    const { error: valueSetError, data: valueSetData } = useQuery<Array<T>>(
```

Comment:
```suggestion
    const { error: valueSetError, data: valueSetData } = useQuery<Array<T>>(
        ["lookupTable", versionData?.version, dataTableName],
        () => getLatestData<Array<T>>(dataTableName, versionData!.version!!),
        { enabled: !!versionData?.version }
    );
```
This didn't throw any issues for me locally. The bang-chain `!.` asserts `versionData` has `version`, and then the trailing double-bang`!!` asserts that `version` is not undefined.

Reply:
as our friends across the pond would say: `brilliant`